app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-enhanced-ecommerce-tracker.js

```diff
@@ -0,0 +1,133 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  'use strict'
+
+  function Ga4EnhancedEcommerceTracker (module) {
+    this.module = module
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.init = function (isNewPage = true) {
+    var consentCookie = window.GOVUK.getConsentCookie()
+    this.isNewPage = isNewPage
+
+    if (consentCookie && consentCookie.settings) {
+      this.startModule()
+    } else {
+      this.startModule = this.startModule.bind(this)
+      window.addEventListener('cookie-consent', this.startModule)
+    }
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.startModule = function () {
+    this.PIIRemover = new window.GOVUK.analyticsGA4.PIIRemover()
+    this.DEFAULT_LIST_TITLE = 'Site search results'
+
+    window.dataLayer = window.dataLayer || []
+
+    if (this.isNewPage) {
+      this.module.addEventListener('click', this.handleClick.bind(this))
+    } else {
+      if (window.GOVUK.analyticsGA4.analyticsModules.PageViewTracker) {
+        var pageViewTracker = window.GOVUK.analyticsGA4.analyticsModules.PageViewTracker
+        window.dataLayer.push({
+          event: 'page_view',
+          page_view: {
+            location: pageViewTracker.getLocation(),
+            referrer: pageViewTracker.getReferrer()
+          }
+        })
+      }
+    }
+
+    this.trackSearchResults()
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.trackSearchResults = function () {
+    var schema = this.populateEcommerceSchema(this.module)
+    this.clearPreviousEcommerceObject()
+    window.dataLayer.push(schema)
+  }
+
+  // the handleClick function will listen for clicks on search results and push to the dataLayer every time a search result is clicked
+  Ga4EnhancedEcommerceTracker.prototype.handleClick = function (event) {
+    var isSearchResult = event.target.getAttribute('data-ecommerce-path')
+
+    if (isSearchResult) {
+      var searchResult = event.target
+      var schema = this.populateEcommerceSchema(true, searchResult)
+
+      this.clearPreviousEcommerceObject()
+      window.dataLayer.push(schema)
+    }
+  }
+
+  Ga4EnhancedEcommerceTracker.prototype.populateEcommerceSchema = function (searchResultClicked = false, searchResult = null) {
+    // Limiting to 100 characters to avoid noise from extra longs search queries
+    // and to stop the size of the payload going over 8k limit.
```

Comment:
Is this is a comment from UA, if so I think it should be removed (the 8k limit part)

Reply:
I think the comment should stay because I think it does relate to GA. Good spot re the typo, I've fixed that now.